web/server/static/js/base.js

```diff
@@ -1,130 +1,174 @@
-function updateTimes() {
-	$('time').each(function(i) {
-		var t = $(this);
-
-		if (t.hasClass("plain")) {return;}
-
-		var dt = t.attr('datetime');
-		if (dt && dt.trim() != '') {
-			var m = moment(dt);
-			var f = t.attr('format');
-			// Relative date for simplicity
-			var td = (f && f.trim() != '') ? m.format(f) : m.fromNow();
-			if (t.text() != td) {
-				t.text(td);
+(() => {
+	// Relative formatter see: https://blog.webdevsimplified.com/2020-07/relative-time-format/
+	const relativeFormatter = new Intl.RelativeTimeFormat(undefined, {
+		numeric: 'auto'
+	})
+
+	const DIVISIONS = [
+		{ amount: 60, name: 'seconds' },
+		{ amount: 60, name: 'minutes' },
+		{ amount: 24, name: 'hours' },
+		{ amount: 7, name: 'days' },
+		{ amount: 4.34524, name: 'weeks' },
```

Comment:
Where did 4.34524 come from?

It looks like `(((365*3+366)/4)/12) / (((365*3+366)/4)/52)` to determine weeks in a month over a year but it's not quite, as that produces 4.3 recurring.

Reply:
Literally from the article I linked in the comment.
Googling, the same factor for month to week conversion seems to be widely used, i.e.:

https://w3percentagecalculator.com/how-many-weeks-in-a-month/
https://www.checkyourmath.com/convert/time/weeks_months.php
https://www.asknumbers.com/DateTimeConversion.aspx

I have no idea where they get the number from and if it's more or less accurate than your formula.

Reply:
LOL, I should read the comment :)

I'm happy to trust the number if the entire world does... I just pause when I see a magic number and see if I can understand it. I think it's roughly what my formula was but probably accounts for it better (i.e. the 4.3 recurring is likely too approximate and someone's stabbed around at the decimal places to make it "good enough" without being too approximate). I cannot find the source of the number used, but hey it works for me.